xdsl/transforms/convert_scf_to_cf.py

```diff
@@ -149,6 +152,67 @@ def match_and_rewrite(self, for_op: For, rewriter: PatternRewriter):
         rewriter.replace_matched_op([], condition_block.args[1:])
 
 
+class SwitchLowering(RewritePattern):
+    """Lowers `scf.index_switch` to `cf.switch`."""
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: IndexSwitchOp, rewriter: PatternRewriter):
+        # Split the block at `op`
+        condition_block = op.parent_block()
+        if condition_block is None:
+            return
+
+        continue_block = condition_block.split_before(op)
+
+        # Create the arguments on the continue block with which to replace the
+        # results of the op.
+        for i, ty in enumerate(op.result_types):
+            rewriter.insert_block_argument(continue_block, i, ty)
+
+        # Handle the regions
+        def convert_region(region: Region) -> Block:
+            block = region.first_block
+            assert block is not None
+
+            # Convert yield op to a branch to the continue block
+            yield_op = block.last_op
+            assert isinstance(yield_op, Yield)
+            rewriter.replace_op(yield_op, Branch(continue_block, *yield_op.operands))
+
+            # Inline the region
+            rewriter.inline_region_before(region, continue_block)
+            return block
```

Comment:
I would recommend outlining this helper so that it's not redefined for every switch operation in the IR